scalanet/src/io/iohk/scalanet/peergroup/InMemoryPeerGroup.scala

```diff
@@ -0,0 +1,142 @@
+package io.iohk.scalanet.peergroup
+
+import monix.eval.Task
+import monix.reactive.{MulticastStrategy, Observable}
+import monix.reactive.subjects.ConcurrentSubject
+import monix.execution.Scheduler.Implicits.global
+import InMemoryPeerGroup._
+
+import scala.collection.concurrent.TrieMap
+import scala.util.Random
+
+// For simplicity we can model the network with three main interactions.
+// 1. Register to the network.
+// 2. Disconnect from the network.
+// 3. Send messages.
+class Network[A, M] {
```

Comment:
Create a separate file unless there is good reason not to.

Reply:
This is internal/relevant only to this PeerGroup. I could attempt to make the Network part of the companion object.